lib/moon/packages/entity_system/entity.rb

```diff
@@ -29,41 +65,81 @@ def add(component, options = {})
         end
       end
 
-      def get_component(key)
+      # Removes a component from the entity
+      #
+      # @param [Array<Component, Symbol>] components
+      def remove(*components)
+        components.each do |component|
+          case component
+          when Component
+            @world.remove_component(self, component)
+          when Symbol
+            @world.remove_component_by_symbol(self, component)
```

Comment:
I think the method on world #remove_component should support both argument types (in case of symbol, look up component first), and then this is simplified